ui/test/feature_ir_test.cpp

```
@@ -0,0 +1,349 @@
+}
+
+}  // namespace
+
+int main() {
+  Harness H("feature_ir");
+
+  // ── 1. the UI table IS the kernel's table ─────────────────────────────────
+  const std::string headerPath = locateKernelHeader();
+  CHECK(!headerPath.empty());  // a gate that cannot find its oracle FAILS
+  bool opened = false;
+  const std::map<std::string, DerivedSpec> kernel = deriveKernelOpTable(headerPath, opened);
```

Clang (14.0.6)

[warning] variable 'kernel' is not initialized

(cppcoreguidelines-init-variables)